src/lib.rs

```diff
@@ -263,23 +262,23 @@ impl Watch {
 #[derive(Debug, StructOpt)]
 pub struct DevServer {
     #[structopt(long, default_value = "127.0.0.1")]
-    ip: IpAddr,
+    pub ip: IpAddr,
     #[structopt(long, default_value = "8000")]
-    port: u16,
+    pub port: u16,
 
     #[structopt(flatten)]
-    watch: Watch,
+    pub watch: Watch,
 }
 
 impl DevServer {
-    pub fn serve(&self, build_dir_path: impl AsRef<Path>) -> Result<()> {
+    pub fn serve(&self, served_path: PathBuf) -> Result<()> {
```

Comment:
I'm not sure why you changed the type here. `respond_to_request` takes a `impl AsRef<Path>`, you should be able to keep it this way.